service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentsController.kt

```diff
@@ -147,9 +147,11 @@ class AttachmentsController(
         val attachment =
             db.read { it.getAttachment(attachmentId) ?: throw NotFound("Attachment $attachmentId not found") }
 
+        val extension = parseExtension(attachment.name) ?: throw Forbidden("Couldn't parse valid file extension for attachment $attachmentId")
+
         return s3Client.get(filesBucket, "$attachmentId").let { document ->
             ResponseEntity.ok()
-                .header("Content-Disposition", "attachment;filename=${document.getName()}")
+                .header("Content-Disposition", "attachment;filename=${document.getName()}.$extension")
```

Comment:
`document.getName()` should probably be replaced with `attachment.name` as well, since it seems kind of silly to have people download files with uuids as their names.